src/vfs/real_fetcher.rs

```diff
@@ -45,7 +45,7 @@ pub struct RealFetcher {
 
     /// All of the paths that the fetcher is watching, tracked here because
     /// notify does not expose this information.
-    watched_paths: HashSet<PathBuf>,
+    watched_paths: Mutex<HashSet<PathBuf>>,
```

Comment:
Was it one Mutex before and now it is two? Does that mean we have to do twice as much locking? Is that ok from a performance perspective?

Are there any tight loops where the performance might be affected by having the mutex on the inside vs outside? And does that matter?

Reply:
This PR just adds the internal mutex, the second one, #260 actually removes the outer mutex and stops us from needing to double-lock.

I branched off for #260 since I was a little less certain about those changes.